src/R/Components/Impl/PackageManager/Implementation/RPackageManager.cs

```diff
@@ -36,11 +52,11 @@ internal class RPackageManager : IRPackageManager {
         }
 
         public async Task<IReadOnlyList<RPackage>> GetInstalledPackagesAsync() {
-            return await GetPackages(async (eval) => await eval.InstalledPackages());
+            return await GetPackagesAsync(async (eval) => await eval.InstalledPackages());
```

Comment:
If the eval uses direct `EvaluateAsync` rather than `BeginEvaluateAsync`, and specifies `REvaluationKind.Reentrant`, it can be interrupted by other evals, allowing to interleave them.